discover/types.go

```diff
@@ -183,3 +183,24 @@ func (si SystemInfo) GetOptimalThreadCount() int {
 
 	return coreCount
 }
+
+// Return if the system supports flash attention
+func (l GpuInfoList) SupportsFlashAttention() bool {
```

Comment:
While most of my testing has been on CUDA/Metal, sometimes run a model that ends up spilling over onto CPU for a few layers - and I found that it does work with some layers offloaded to CPU:

```
llm_load_tensors: offloading 68 repeating layers to GPU
llm_load_tensors: offloaded 68/81 layers to GPU
llm_load_tensors:        CPU buffer size = 37669.41 MiB
llm_load_tensors:      CUDA0 buffer size = 15378.36 MiB
llm_load_tensors:      CUDA1 buffer size = 15378.36 MiB
llama_new_context_with_model: n_ctx      = 32768
llama_new_context_with_model: n_batch    = 512
llama_new_context_with_model: n_ubatch   = 512
llama_new_context_with_model: flash_attn = 1
llama_kv_cache_init:  CUDA_Host KV buffer size =   816.00 MiB
llama_kv_cache_init:      CUDA0 KV buffer size =  2312.00 MiB
llama_kv_cache_init:      CUDA1 KV buffer size =  2312.00 MiB
llama_new_context_with_model: KV self size  = 5440.00 MiB, K (q8_0): 2720.00 MiB, V (q8_0): 2720.00 MiB
```

Reply:
Got it, thanks. Should we change this check then to be inclusive of CPUs?

Reply:
I _think_ for now I'd leave it as the GPUs, I'm really hesitant to introduce too many moving pieces in one PR.

Would you be ok with looking to roll out out (if desired) in a future PR? That will give folks a chance to get a feel for it without bringing the performance issues that CPU inference (in general) has?

Reply:
Sounds good to me. The one issue I can see with it as-is is that it will be enabled even with CPU offload. As long as that still works I'm OK with this

Reply:
Yeah I haven't had any issues with CPU offload personally, I've added a few extra test cases for mixed offloading so if there ever are changes to behaviour they should be caught.

Reply:
Fixed this up, I don't know if I was half asleep when I thought that FA mixed GPU support logic was correct, what a facepalm moment.

Reply:
Looking much better